### ISSUE TITLE
Finetuning

### ISSUE THREAD

**Issue by VanitarNordic** [CharlesShang/FastMaskRCNN#41]
Hi,

I don't want to retrain the model from scratch, I want to fine-tune it using a different dataset and most likely different number of classes. is there anyway to do so?

**Comment by kevinkit** [comment]
See #34 , #31

**Comment by VanitarNordic** [comment]
Thank you but they are about training from scratch, not fine-tuning.

**Comment by kevinkit** [comment]
For fine-tuning you will need pretrained models  and someone who has succesfully trained it, both issues show that this is not the case, thus it is not yet possible to do finetuning on different datasets due to the lack of pretrained models

**Comment by VanitarNordic** [comment]
if a pre-trained model exist in `.caffemodel` format, it can be converted (potentially) to tensorflow. The only thing remain would be some modifications in the model to accept a different number of class. I think it should be done with last conv layer generally, but I''m not quite sure about MaskRCNN. You are tensorflow expert, you know more.

**Comment by kevinkit** [comment]
Till now and to my knowledge there is no implementation of the Mask RCNN in any framework available. While the paper states that 

> code will be made availble 

[Mask RCNN paper](https://arxiv.org/abs/1703.06870)

it has not been done yet - and if so it is likely to be in Torch or similiar since it comes from FAIRE. 

However if you are looking for a way to already have a trained model, you could try to train the Faster R-CNN with your dataset and wait till this repository is finished and then try to merge both architectures. MASK-RCNN is an extension of Faster RCNN.
